src/applications/claims-status/actions/index.jsx

```diff
@@ -147,16 +148,55 @@ export function fetchClaimsSuccess(response) {
   };
 }
 
-export function getClaimsV2() {
+const POLLING_INTERVAL = 1000;
+
+export function pollClaimsStatus({
+  onError,
+  onSuccess,
+  pollingInterval,
+  request = apiRequest,
+  shouldFail,
+  shouldSucceed
+}) {
+  return request(
+    '/evss_claims_async',
+    null,
+    response => {
+      if (shouldSucceed(response)) {
+        onSuccess(response);
+        return;
+      }
+
+      if (shouldFail(response)) {
```

Comment:
Do we anticipate wanting to configure shouldFail and shouldSucceed? If not, could we pull this logic into a checkStatus helper to house these checks, as well as the call to pollClaimsStatus (Similar to the burial app)?

Reply:
I plan on reusing this for the appeals status. Could have gone with a switch statement- this is mostly a style decision on my part to invert the control so it's easier to tell what's going on without having to read the polling function. I figure a maintainer will get to this from the action name or the action creator call, and they can tell what this is doing just by reading the arguments. 

![screen shot 2018-08-20 at 13 53 04](https://user-images.githubusercontent.com/4998130/44363126-6d38ac80-a480-11e8-960e-bcabf87dea87.png)

Reply:
Thanks for the context, sounds good.

Reply:
Since we are planning to reuse this for appeals, will it have the same shouldFail, and shouldSucceed logic? If so, could that simply be specified, rather than require configuration?

Reply:
I think the extra boilerplate is fine- I can read the `poll` function's arguments and tell what its doing without reading its implementation. If I push the predicates into the polling function then I know it's "polling" but I don't know when it's going to call success or failure without looking at its implementation. 

Dunno if this is a legitimate design approach, though- it's inversion of control to me.